<a href="https://colab.research.google.com/github/Satyake/Shopily-Challenge-Link/blob/main/Shopify_DS_Challenge.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np 
import matplotlib.pyplot as plt


In [2]:
data=pd.read_excel('/content/2019 Winter Data Science Intern Challenge Data Set.xlsx')

In [3]:
data.head()    

order_id  shop_id  ...  payment_method              created_at
0         1       53  ...            cash 2017-03-13 12:36:56.190
1         2       92  ...            cash 2017-03-03 17:38:51.999
2         3       44  ...            cash 2017-03-14 04:23:55.595
3         4       18  ...     credit_card 2017-03-26 12:43:36.649
4         5       18  ...     credit_card 2017-03-01 04:35:10.773

[5 rows x 7 columns]

In [4]:
data.describe()

order_id      shop_id      user_id   order_amount  total_items
count  5000.000000  5000.000000  5000.000000    5000.000000   5000.00000
mean   2500.500000    50.078800   849.092400    3145.128000      8.78720
std    1443.520003    29.006118    87.798982   41282.539349    116.32032
min       1.000000     1.000000   607.000000      90.000000      1.00000
25%    1250.750000    24.000000   775.000000     163.000000      1.00000
50%    2500.500000    50.000000   849.000000     284.000000      2.00000
75%    3750.250000    75.000000   925.000000     390.000000      3.00000
max    5000.000000   100.000000   999.000000  704000.000000   2000.00000

In [22]:
order_amount

array([224,  90, 144, ..., 351, 354, 288])

**To Calculate the Average:**

In [11]:
order_amount=data.iloc[:,3].values
total_items=data.iloc[:,4].values

In [8]:
order_amount_sum=np.sum(order_amount)
print(order_amount_sum)

15725640


In [12]:
total_amounts_sum=np.sum(total_items)
print(total_amounts_sum)

43936


In [14]:
round(order_amount_sum/total_amounts_sum,2)

357.92

**#357.92 is the correct average  going by the order amounts**

In [26]:
len(total_items)
order_amount_sum/len(total_items)



3145.128

**Issue is the followng If we count the total items instead of adding then we are getting the incorrect answer
calling the length function over total items lead to the incorrect result.
it is a very simple mistake that may have happened when the person was working with the data**